sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -3,27 +3,89 @@
 
 #pragma once
 
+#include <string>
+#include <vector>
+
 #include <internal/contract.hpp>
 
+using namespace std;
+
 namespace azure
 {
 namespace core
 {
 namespace http
 {
 
-class http_request
+namespace http_method
 {
+enum HttpMethod
+{
+  GET,
+  HEAD,
+  POST,
+  PUT,
+  DELETE,
+  PATCH,
+};
+}
 
+class Header
+{
 private:
-  /* data */
+  string name;
+  string value;
+
 public:
-  http_request();
-  ~http_request();
+  Header(string name, string value)
```

Comment:
you can initialize it as `:name(name), value(value)`.
Since the name is the same you technicallycan't, but I believe we should have some prefix/suffix naming convention for data members. Such as `m_` prefix or `_` suffix.

Reply:
I like the `_` prefix

Reply:
`: name(name), value(value)` is grammatically correct by C++ standard, the compiler is smart enough to tell which is member variable and which is argument.

But we should write `: name(std::move(name)), value(std::move(value))` to avoid making another copy.

Reply:
well, I removed Header
But, can you take a look to Request Constructors?